openapi/tests/openapi_integration/test_shard_snapshot.py

```diff
@@ -0,0 +1,165 @@
+        path_params={'shard_id': 0, 'collection_name': collection_name},
+        query_params={'wait': 'false'},
+    )
+    assert response.status_code == 202
+
+    # validate it exists
+    snapshot_name = None
+    while True:
+        try:
+            response = request_with_validation(
+                api='/collections/{collection_name}/shards/{shard_id}/snapshots',
+                method="GET",
+                path_params={'shard_id': 0, 'collection_name': collection_name},
+            )
+            assert response.ok
+            assert len(response.json()['result']) == 1
+            snapshot_name = response.json()['result'][0]['name']
+            break
+        except AssertionError:
+            # wait for snapshot to be created
+            sleep(0.1)
+            continue
+
+    # delete it
+    response = request_with_validation(
+        api='/collections/{collection_name}/shards/{shard_id}/snapshots/{snapshot_name}',
+        method="DELETE",
+        path_params={'shard_id': 0, 'collection_name': collection_name,
+                     'snapshot_name': snapshot_name},
+        query_params={'wait': 'false'},
+    )
+    assert response.status_code == 202
+
+    # validate it is gone
+    while True:
+        try:
+            response = request_with_validation(
+                api='/collections/{collection_name}/shards/{shard_id}/snapshots',
+                method="GET",
+                path_params={'shard_id': 0, 'collection_name': collection_name},
+            )
+            assert response.ok
+            assert len(response.json()['result']) == 0
+            break
+        except AssertionError:
+            # wait for snapshot to be deleted
+            sleep(0.1)
+            continue
+
+
+def test_snapshot_invalid_file_uri():
+    # Invalid file:// host
+    response = request_with_validation(
+        api='/collections/{collection_name}/shards/{shard_id}/snapshots/recover',
+        method="PUT",
+        path_params={'shard_id': 0, 'collection_name': "somethingthatdoesnotexist"},
+        body={
+            "location": "file://whatever.snapshot",
+        }
+    )
+    assert response.status_code == 404
+
+    # Absolute path that does not exist
+    response = request_with_validation(
+        api='/collections/{collection_name}/shards/{shard_id}/snapshots/recover',
+        method="PUT",
+        path_params={'shard_id': 0, 'collection_name': "somethingthatdoesnotexist"},
+        body={
+            "location": "file:///whatever.snapshot",
+        }
+    )
+    assert response.status_code == 404
+
+    # Path that does not exist
+    response = request_with_validation(
+        api='/collections/{collection_name}/shards/{shard_id}/snapshots/recover',
+        method="PUT",
+        path_params={'shard_id': 0, 'collection_name': "somethingthatdoesnotexist"},
+        body={
+            "location": "file://localhost/whatever.snapshot",
+        }
+    )
+    assert response.status_code == 404
```

Comment:
We decided to not use `file://` URLs for shard snapshot recovery, and so explicitly check that `locations` is either HTTP(S) URL or a _plain path_. `file://` should return error 400.

See:
- https://github.com/qdrant/qdrant/blob/dev/src/actix/api/snapshot_api.rs#L334-L341
- https://github.com/qdrant/qdrant/blob/dev/src/actix/api/snapshot_api.rs#L557-L562